src/goose.rs

```diff
@@ -965,8 +961,6 @@ impl GooseUser {
             weighted_on_start_tasks: Vec::new(),
             weighted_tasks: Vec::new(),
             weighted_on_stop_tasks: Vec::new(),
-            task_request_name: None,
-            request_name: None,
```

Comment:
Oddly `request_name` wasn't even used. And `task_request_name` wasn't consistently working correctly as the required information wasn't directly available to the `GooseUser`. Instead we expand the `WeightedGooseTasks` type to include the GooseTask name, if set.